scheme/lib/core.sld

```diff
@@ -275,7 +275,7 @@
                      ((func? x)
                       (let ((func (make-func (func-ast x) (func-params x)
                                              (func-env x) (func-fn x))))
-                        (func-macro?-set! func (func-macro? x))
+                        (func-macro?-set! #f)
```

Comment:
Why would you do that?

Reply:
This PR intends to test the quasiquote changes. I have based it on #384 and #400 for convenience, but related discussions should happen at #384.
Anyway...
- this first change is replaced with a more correct ```(func-macro?-set! func #f)``` by a following commit.
- The way Guile and Scheme support metadata for macros breaks macros in the MAL implementation in #384. I have removed this optional feature, and introduced a soft test in case someone wants to work on a real fix. The intent is that a broken optional feature does not block the merge of #384.
However, #400 implements macros in MAL without metadata. If it is accepted, we could as well revert the changes in Guile and Scheme.

Reply:
I see, thanks.